tianshou/data/buffer.py

```diff
@@ -151,6 +151,8 @@ def _add_to_buffer(self, name: str, inst: Any) -> None:
 
     def update(self, buffer: 'ReplayBuffer') -> None:
         """Move the data from the given buffer to self."""
+        if len(buffer) == 0:
+            return
```

Comment:
bugfix for update with empty buffer

Reply:
Why not raise an exception ? Is doing this allowed ?

Reply:
Update a buffer with another empty buffer is allowed. Just like extending a list with another list ``a += b``. We just need to assure it behaves correctly.